### PR TITLE
Dark mode implementation

### PR DESCRIPTION
**Pull request by SzymonGorowski**
https://trello.com/c/pr99AipA/696-introduce-darkmode-to-the-ios-app

I am aware that the image on the first onboarding screen is stretched but we need to ask Gosia to provide a bigger one because now SwiftUI stretching images to fit screen sizes on bigger devices.

Also, I am aware that the map flashes with white color when appears on the screen.

### CODE REVIEW

File: AirCasting/CreateSessionViews/ChooseSessionType/ChooseSessionTypeView.swift [hunk 1]
```diff
@@ -263,12 +262,10 @@ struct ChooseSessionTypeView: View {
...
             .padding([.bottom, .vertical])
             .padding(.horizontal, 30)
-            .background(
-                Color.aircastingBackground.opacity(0.25)
-                    .ignoresSafeArea()
-            )
+            .background(Color.aircastingSecondaryBackground.ignoresSafeArea())
             .alert(item: $viewModel.alert, content: { $0.makeAlert() })
         }
+        .background(Color.aircastingBackgroundWhite.ignoresSafeArea())
```
Comment (jankrzempek) [inline AirCasting/CreateSessionViews/ChooseSessionType/ChooseSessionTypeView.swift:268]: Do you need this one?

File: AirCasting/CreateSessionViews/PowerABView.swift [hunk 2]
```diff
@@ -30,9 +30,8 @@ struct PowerABView: View {
                 .buttonStyle(BlueButtonStyle())
         }
         .padding()
-        .onAppear(perform: {
-            sessionContext.deviceType = .AIRBEAM3
-        })
+        .onAppear(perform: { sessionContext.deviceType = .AIRBEAM3 })
```
Comment (jankrzempek) [inline AirCasting/CreateSessionViews/PowerABView.swift:33]: 🙌🏻

File: AirCasting/SearchAndFollow/CompleteScreen/Views/CompleteScreenViewModel.swift [hunk 3]
```diff
@@ -82,11 +82,15 @@ class CompleteScreenViewModel: ObservableObject {
     }
     
     func mapTapped() {
-        isMapSelected = true
+        withAnimation {
+            isMapSelected = true
+        }
```
Comment (jankrzempek) [inline AirCasting/SearchAndFollow/CompleteScreen/Views/CompleteScreenViewModel.swift:87]: One liner maybe? What do you prefer 🫡

File: AirCasting/SearchAndFollow/CompleteScreen/Views/CompleteScreenViewModel.swift [hunk 4]
```diff
@@ -82,11 +82,15 @@ class CompleteScreenViewModel: ObservableObject {
...
     }
     
     func chartTapped() {
-        isMapSelected = false
+        withAnimation {
+            isMapSelected = false
+        }
```
Comment (jankrzempek) [inline AirCasting/SearchAndFollow/CompleteScreen/Views/CompleteScreenViewModel.swift:93]: One liner maybe? What do you prefer 🫡

**Review by jankrzempek — APPROVED** [review]
That was HUGE! This is the way 🫡
Consider few comments, please

File: AirCasting/CreateSessionViews/ChooseSessionType/ChooseSessionTypeView.swift (hunk 1 above)
Comment (SzymonGorowski) [reply AirCasting/CreateSessionViews/ChooseSessionType/ChooseSessionTypeView.swift:268]: Yes, I need this one for the background on the top as on the attached image.

Simulator Screen Shot - iPhone 11 - 2022-07-26 at 09 36 31

File: AirCasting/SearchAndFollow/SearchMap/SearchMapView.swift [hunk 5]
```diff
@@ -40,12 +41,12 @@ struct SearchMapView: View {
                                     .padding(.bottom, 28)
                             } else {
                                 ZStack(alignment: .topLeading) {
-                                    LinearGradient(gradient: Gradient(colors: [.white.opacity(0.1),
-                                                                               .white.opacity(0.5),
-                                                                               .white.opacity(0.7),
-                                                                               .white.opacity(0.8),
-                                                                               .white.opacity(0.9),
-                                                                               .white]),
+                                    LinearGradient(gradient: Gradient(colors: [.gradientColor.opacity(0.1),
```
Comment (anna1901) [inline AirCasting/SearchAndFollow/SearchMap/SearchMapView.swift:44]: I think it would be better to use our custom background color here instead

**Review by anna1901 — CHANGES_REQUESTED** [review]
What a great work!! 👏🏻👏🏻👏🏻
I have some minor suggestions for changes. Additionally, there in one thing left which probably could be changed to dark mode and it's the WebViewController which we are using for displaying privacy policy.

File: AirCasting/Onboarding/GetStarted.swift [hunk 6]
```diff
@@ -53,31 +66,27 @@ private extension GetStarted {
...
-            .background(
-                RoundedRectangle(cornerRadius: 5)
-                    .stroke(lineWidth: 0.2)
-                    .accentColor(Color.aircastingGray)
-            )
-            .padding()
-            .padding(.bottom, 41)
+        .buttonStyle(BlueButtonStyle())
+        .padding()
+        .padding(.bottom, 42)
```
[the remark this answers is not in the document]
Comment (SzymonGorowski) [reply AirCasting/Onboarding/GetStarted.swift:76]: I think now is more suitable :D

File: AirCasting/Dashboard/SessionListView.swift [hunk 7]
```diff
@@ -83,7 +83,7 @@ struct SessionsListView: View {
                 }
             }
             .coordinateSpace(name: listCoordinateSpaceName)
-            .background(Color.aliceBlue)
+            .background(Color.aircastingWhite)
```
Comment (anna1901) [inline AirCasting/Dashboard/SessionListView.swift:86]: Can this `aircastingWhite` color be replaced with `secondaryBackground`? Aren't they the same?

**Review by anna1901 — APPROVED** [review]
Just a small comment for you to consider. Really great job Szymon 😊👏🏻🚀

File: AirCasting/Dashboard/SessionListView.swift (hunk 7 above)
Comment (anna1901) [reply AirCasting/Dashboard/SessionListView.swift:86]: (I mean replacing it and deleting in general, not just here)

Comment (SzymonGorowski) [reply AirCasting/Dashboard/SessionListView.swift:86]: Unfortunately, these colors are different a little bit.

**Pull request merged**